include/IECore/Version.h

```diff
@@ -0,0 +1,97 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2020, Image Engine Design Inc. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+//     * Neither the name of Image Engine Design nor the names of any
+//       other contributors to this software may be used to endorse or
+//       promote products derived from this software without specific prior
+//       written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#ifndef IECORE_VERSION_H
+#define IECORE_VERSION_H
+
+#define CORTEX_MILESTONE_VERSION IE_CORE_MILESTONEVERSION
+#define CORTEX_MAJOR_VERSION IE_CORE_MAJORVERSION
+#define CORTEX_MINOR_VERSION IE_CORE_MINORVERSION
+#define CORTEX_PATCH_VERSION IE_CORE_PATCHVERSION
+
+#define MAKE_CORTEX_COMPATIBILITY_VERSION( MILESTONE_VERSION, MAJOR_VERSION ) \
+	( MILESTONE_VERSION * 1000 + MAJOR_VERSION  )
+
+#define CORTEX_COMPATIBILITY_VERSION ( MAKE_CORTEX_COMPATIBILITY_VERSION( CORTEX_MILESTONE_VERSION, CORTEX_MAJOR_VERSION ) )
+
+namespace IECore
+{
+
+/// Returns the milestone version for the IECore library
+IECORE_API inline int milestoneVersion()
```

Comment:
If we're breaking compatibility by moving these to a different header, is there any merit in just removing them since we now have the macros? Personally I'm more inclined to put them back where they were though, and leave the compatibility breaks for later, once the versioning change has settled in.

Reply:
> is there any merit in just removing them since we now have the macros

They're used differently than the macros though, right? The macros are for compile time logic and the methods are for runtime logic. Admittedly, I've only ever used them via the bindings in python, but I guess someone could want runtime support for multiple compatibility versions in c++ too.

> Personally I'm more inclined to put them back where they were though, and leave the compatibility breaks for later,

I'd done that originally and then found it a bit odd that Version.h didn't include them... I can put it back if you feel its important to maintain the compatibility though. Note its still a breaking change because `IECore::majorVersion()` still has a totally different meaning and any runtime logic in dependent projects will certainly need updating.

Reply:
I haven't moved these back to IECore.h (yet). Do you think I should or its ok to leave them in Version.h ?

Reply:
Version.h is clearly the logical place. But the concern would be that code needing conditional compilation both sides of the version divide wouldn't even know if it could `#include "IECore/Version.h"`, because it wouldn't exist on the previous side. I don't think we'll need that for Gaffer though. I've also learned that there's a [`__has_include()`](https://stackoverflow.com/a/33260104) preprocessor directive that is available for at least some compilers and versions. Maybe that's enough? If not, I suggest a compromise : put it in `Version.h` but also `#include "IECore/Version.h"` from `IECore.h`.

Reply:
Sure thing. We won't need that safety net at IE either, but no harm putting it in there.

Reply:
Addressed in 83f696390f153cacb9f94992bfb4b95e88d475a8